language/move-prover/bytecode/src/read_write_set_analysis.rs

```diff
@@ -879,3 +842,67 @@ impl Default for ReadWriteSetState {
         }
     }
 }
+
+// =================================================================================================
+// Converting Infer Result into standalone types
+
+impl Offset {
+    pub fn make_canonical(&self, env: &GlobalEnv) -> Option<RWOffset> {
+        Some(match self {
+            Offset::Field(idx) => RWOffset::Field(*idx),
+            Offset::VectorIndex => RWOffset::VectorIndex,
+            Offset::Global(s) => RWOffset::Global(s.get_type().into_struct_type(env)?),
+        })
+    }
+}
+impl AccessPath {
+    pub fn make_canonical(&self, env: &GlobalEnv) -> Option<Vec<RWAccessPath>> {
+        let (roots, access_opt) = match self.root() {
+            Root::Formal(idx) => (vec![RWRoot::Formal(*idx)], None),
+            Root::Global(key) => (
+                key.address()
+                    .get_concrete_addresses()
+                    .into_iter()
+                    .map(RWRoot::Const)
+                    .collect(),
+                key.struct_type().get_type().into_struct_type(env),
+            ),
+            _ => return None,
+        };
+        let mut offsets = access_opt
+            .clone()
+            .map_or_else(|| vec![], |struct_ty| vec![RWOffset::Global(struct_ty)]);
+
+        offsets.append(
+            &mut self
+                .offsets()
+                .iter()
+                .map(|offset| offset.make_canonical(env))
+                .collect::<Option<Vec<_>>>()?,
+        );
+
+        Some(
+            roots
+                .into_iter()
+                .map(|root| RWAccessPath {
+                    root,
+                    offsets: offsets.clone(),
+                })
+                .collect(),
+        )
+    }
+}
+
+impl ReadWriteSetState {
+    pub fn make_canonical(&self, env: &GlobalEnv) -> Option<ReadWriteSet> {
+        let mut analysis_result = ReadWriteSet::new();
+        self.accesses.iter_paths(|access_path, access| {
+            // TODO: Replace the unwrap here?
+            let access_pathes = access_path.make_canonical(env).unwrap();
```

Comment:
Yeah, seems like we should return `None` in this case. Otherwise, the function always returns `Some`

Reply:
Fixed